tests/json_common_interface_test.rb

```diff
@@ -123,4 +123,61 @@ def test_JSON
     assert_equal @json, JSON(@hash)
     assert_equal @hash, JSON(@json)
   end
+
+
+  def test_load_file
+    test_load_shared(:load_file)
+  end
+
+  def test_load_file!
+    test_load_shared(:load_file!)
+  end
+
+  def test_load_file_with_option
+    test_load_file_with_option_shared(:load_file)
+  end
+
+  def test_load_file_with_option!
+    test_load_file_with_option_shared(:load_file!)
+  end
+
+  private
+
+  def test_load_shared(method_name)
+    temp_file_containing(@json) do |filespec|
+      assert_equal method(method_name).call(filespec), @hash
```

Comment:
`method(method_name).call(filespec)` => `public_send(method_name, filespec)`. I think it's more idiomatic.
Same for `test_load_file_with_option_shared`

Reply:
I've made this change and squashed the commits. Thanks for your guidance, I agree that these are both improvements.

Reply:
Thanks for the impressively quick turnaround and good PR!